packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -159,6 +167,12 @@ export abstract class SharedObjectCore<
 		});
 		this.mc = loggerToMonitoringContext(this.logger);
 
+		const maybeRuntimeCompatDetails = this.runtime as FluidObject<ILayerCompatDetails>;
+		this.skipHandleEncodingOnSubmit =
+			maybeRuntimeCompatDetails?.ILayerCompatDetails?.supportedFeatures.has(
```

Comment:
Oops this is wrong.  This is the DataStore supported features it's telling the CR about.  I need to expose the DataStoreContext's ILayerCompatDetails somehow.  @agarwal-navin let's discuss, probably next week.

I'm going to pull that into a separate PR too so we can discuss but not block this first step, I want to keep working on the next step in the CR, this can wait a minute.